src/components/QueryRecordings/QueryRecordings.vue

```diff
@@ -115,246 +94,168 @@ export default {
-          target.where.DeviceId = [
-            ...target.where.DeviceId,
-            ...target.where.DeviceGroups
-          ];
-        } else if (target.where.DeviceGroups) {
-          target.where.DeviceId = [...target.where.DeviceGroups];
-        }
+    },
+    makeArray(value) {
+      if (typeof value === "object") {
+        return value;
+      } else {
+        return value ? [value] : [];
       }
-      if (routeQuery.tags) {
-        target.tags = JSON.parse(routeQuery.tags);
+    },
+    setAdvancedInitalState() {
+      // If there was an advanced query, start with the advanced toggle area open.
+      this.advanced =
+        this.hasTagData() ||
+        this.duration.hasOwnProperty("maxS") ||
+        this.duration.hasOwnProperty("minS");
+    },
+    hasTagData() {
+      return (
+        this.tagMode !== "any" ||
+        (this.tagData.tags && this.tagData.tags.length > 1)
+      );
+    },
+    deserialiseRouteIntoQuery(routeQuery) {
+      this.setOnlyIfExists("offset", routeQuery, this.query);
+      this.setOnlyIfExists("limit", routeQuery, this.query);
+
+      this.setOnlyIfExists("tagMode", routeQuery, this.tagData);
+
+      this.setOnlyIfExists("minS", routeQuery, this.duration);
+      this.setOnlyIfExists("maxS", routeQuery, this.duration);
+
+      if (routeQuery.hasOwnProperty("tag")) {
+        this.tagData.tags = this.makeArray(routeQuery.tag);
+      }
+
+      if (routeQuery.hasOwnProperty("type")) {
+        this.recordingType = routeQuery.type;
+      }
+
+      this.dates = {
+        days: routeQuery.days,
+        to: routeQuery.to,
+        from: routeQuery.from
+      };
+
+      if (routeQuery.hasOwnProperty("group")) {
+        this.selectedGroups = this.makeArray(routeQuery.group);
+      }
+      if (routeQuery.hasOwnProperty("device")) {
+        this.selectedDevices = this.makeArray(routeQuery.device);
       }
+
+      this.setAdvancedInitalState();
     },
-    parseCurrentRoute(addState = true) {
-      if (Object.keys(this.$route.query).length === 0) {
-        // Populate the url params if we got here without them, ie. /recordings
-        this.query = this.resetQuery();
-        if (addState) {
-          this.updateRouteQuery();
+    serialiseQueryForRecall() {
+      const params = {
+        tagMode: this.tagData.tagMode,
+        minS: this.duration.minS,
+        maxS: this.duration.maxS,
+        tag: this.tagData.tags,
+        limit: this.query.limit,
+        offset: this.query.offset,
+        days: this.dates.days,
+        from: this.dates.from,
+        to: this.dates.to,
+        type: this.recordingType
+      };
+
+      if (this.selectedDevices.length > 0) {
+        params.device = this.selectedDevices;
+      }
+
+      if (this.selectedGroups.length > 0) {
+        params.group = this.selectedGroups;
+      }
+
+      // remove undefined elements
+      for (const key in params) {
+        if (!params[key] || params[key] === []) {
```

Comment:
Using triple-equals `===` params[key] will never be equal to [], since triple equals is based on the object reference, not its content.  In this instance, double equals wouldn't ever be true either, so I guess what you want is just to check if the length is zero?

Reply:
Also, it's better not to do `!params[key]` as a check, since if `key` is set on `params`, but the value is falsey, this will return true, which is probably not what you want.  Better to use `params.hasOwnProperty(key)`